client/components/dashboard/dashboard-service.js

```diff
@@ -416,17 +417,17 @@ DashboardService.prototype.selectWidget = function(
     }
   }
 
-  if (widget) {
+  if (goog.isDefAndNotNull(widget)) {
     this.sidebarTabService_.resolveSelectedTabForWidget();
 
     this.timeout_(() => {
-      this.scrollWidgetIntoView(widget);
+      this.scrollWidgetIntoView(/** @type {!WidgetConfig} */ (widget));
```

Comment:
That was my assumption as well, but the timeout_() anonymous function seems to confuse the compiler's optimizer.  I can either duplicate the IsNull check or cast it (which is safe, since I've already done the check).